gammapy/scripts/cta_irf.py

```diff
@@ -392,24 +393,42 @@ def read(cls, filename, offset='0.5 deg'):
         ----------
         filename : `str`
             File containing the IRFs
+        e_reco : `~astropy.units.Quantity` array
+            Array of energies to reproject the background rate
         """
         filename = str(make_path(filename))
-
+        
         hdulist = fits.open(filename)
         aeff = EffectiveAreaTable.from_hdulist(hdulist=hdulist)
         edisp = EnergyDispersion2D.read(filename, hdu='ENERGY DISPERSION')
-        bkg = BgRateTable.from_hdulist(hdulist=hdulist)
         psf = Psf68Table.from_hdulist(hdulist=hdulist)
         sens = SensitivityTable.from_hdulist(hdulist=hdulist)
 
-        # Create rmf with appropriate dimensions (e_reco->bkg, e_true->area)
-        e_reco_min = bkg.energy.lo[0]
-        e_reco_max = bkg.energy.hi[-1]
-        e_reco_bin = bkg.energy.nbins
-        e_reco_axis = EnergyBounds.equal_log_spacing(
-            e_reco_min, e_reco_max, e_reco_bin, 'TeV',
-        )
+        # Reproject background if needed
+        e_reco_axis = None
+        bkg = None
+        if e_reco is None:
+            bkg = BgRateTable.from_hdulist(hdulist=hdulist)
+            e_reco_min = bkg.energy.lo[0]
+            e_reco_max = bkg.energy.hi[-1]
+            e_reco_bin = bkg.energy.nbins
+            e_reco_axis = EnergyBounds.equal_log_spacing(
+                e_reco_min, e_reco_max, e_reco_bin, 'TeV',
+            )
+        else:
+            bkg_e = BgRateTable.from_hdulist(hdulist=hdulist)  # 1 / s
+            bkg_e.data.data = bkg_e.data.data / bkg_e.energy.bin_width  # 1 / (s TeV)
+            e_reco_axis = EnergyBounds(e_reco)
+            vals = bkg_e.data.evaluate(energy=e_reco_axis.log_centers,
+                                       method = 'linear')
+            idx = np.where(np.isfinite(vals))[0]
```

Comment:
I think this can probably be rewritten in a simpler way using a bool mask?
```
mask = np.isfinite(vals)
```
Can you have a look, or at least try to establish the desired behaviour via an assert on the content of bkg in the test? (and then I could try to rewrite these three lines that are hard to understand)